src/include/distributed/distributed_transaction_management.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * distributed_transaction_management.h
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef DISTRIBUTED_TRANSACTION_MANAGEMENT_H_
+#define DISTRIBUTED_TRANSACTION_MANAGEMENT_H_
+
+
+#include "datatype/timestamp.h"
+#include "nodes/pg_list.h"
+
+
+/*
+ * Citus identifies a distributed transaction with a triplet consisting of
+ *
+ *  -  initiatorNodeIdentifier: A unique identifier of the node that initiated
+ *     the distributed transaction
+ *  -  transactionId: A locally unique identifier assigned for the distributed
+ *     transaction on the node that initiated the distributed transaction
+ *  -  timestamp: The current timestamp of distributed transaction initiation
+ *
+ */
+typedef struct DistributedTransactionId
+{
+	uint64 initiatorNodeIdentifier;
+	uint64 transactionId;
+	TimestampTz timestamp;
+} DistributedTransactionId;
+
+
+/*
+ * Each backend's active distributed transaction information is tracked via
+ * DistributedTransactionBackendData on the shared memory.
+ */
+typedef struct DistributedTransactionBackendData
+{
+	Oid databaseId;
+	DistributedTransactionId transactionId;
+} DistributedTransactionBackendData;
```

Comment:
The spinlock should live here, per-backend, rather than in process local memory.